flocker/provision/_install.py

```diff
@@ -256,43 +320,96 @@ def task_install_flocker(
         distribution=None,
         package_source=PackageSource()):
     """
-    Install flocker.
+    Install flocker on a distribution.
 
     :param bytes distribution: The distribution the node is running.
     :param PackageSource package_source: The source from which to install the
         package.
     """
-    commands = [
-        run(command="yum install -y " + ZFS_REPO[distribution]),
-        run(command="yum install -y " + CLUSTERHQ_REPO[distribution])
-    ]
-
     if package_source.branch:
         result_path = posixpath.join(
             '/results/omnibus/', package_source.branch, distribution)
         base_url = urljoin(package_source.build_server, result_path)
-        repo = dedent(b"""\
-            [clusterhq-build]
-            name=clusterhq-build
-            baseurl=%s
-            gpgcheck=0
-            enabled=0
-            """) % (base_url,)
-        commands.append(put(content=repo,
-                            path='/etc/yum.repos.d/clusterhq-build.repo'))
-        branch_opt = ['--enablerepo=clusterhq-build']
     else:
-        branch_opt = []
-
-    if package_source.os_version:
-        package = 'clusterhq-flocker-node-%s' % (package_source.os_version,)
+        base_url = None
+    if distribution == 'ubuntu-14.04':
+        commands = [
+            # Ensure add-apt-repository command is available
+            run_from_args([
+                "apt-get", "-y", "install", "software-properties-common"]),
+            # ZFS not available in base Ubuntu - add ZFS repo
+            run_from_args([
+                "add-apt-repository", "-y", "ppa:zfs-native/stable"]),
+            # Add Docker repo for recent Docker versions
+            run_from_args([
+                "add-apt-repository", "-y", "ppa:james-page/docker"]),
+            # Add ClusterHQ repo for installation of Flocker packages.
+            run_from_args([
+                'add-apt-repository', '-y',
+                'deb https://s3.amazonaws.com/clusterhq-archive/ubuntu 14.04/amd64/'  # noqa
+                ])
+            ]
+
+        if base_url:
+            # Add BuildBot repo for testing
+            commands.append(run_from_args([
+                "add-apt-repository", "-y", "deb {} /".format(base_url)]))
+
+        commands += [
+            # Update to read package info from new repos
+            run_from_args([
+                "apt-get", "update"]),
+            # Package spl-dkms sometimes does not have libc6-dev as a
+            # dependency, add it before ZFS installation requires it.
+            # See https://github.com/zfsonlinux/zfs/issues/3298
```

Comment:
This ticket is closed as a dupe of https://github.com/zfsonlinux/pkg-zfs/issues/129 which is itself closed and looks like it won't be re-opened.